static/app/components/profiling/flamegraph.tsx

```diff
@@ -26,7 +26,7 @@ function Flamegraph(props: FlamegraphProps): React.ReactElement {
   const canvasPoolManager = useMemo(() => new CanvasPoolManager(), []);
 
   const [flamegraph, setFlamegraph] = useState(
-    new FlamegraphModel(props.profiles.profiles[0], 0, {
+    new FlamegraphModel(props.profiles.profiles[props.profiles.activeProfileIndex], 0, {
       inverted: view === 'bottom up',
       leftHeavy: sorting === 'left heavy',
     })
```

Comment:
Is the second argument to `FlamegraphModel` used anywhere?

```suggestion
    new FlamegraphModel(
      props.profiles.profiles[props.profiles.activeProfileIndex],
      props.profiles.activeProfileIndex,
      {
        inverted: view === 'bottom up',
        leftHeavy: sorting === 'left heavy',
      }
    )
```

Also, I just noticed but why is there an `useEffect` further down always uses the 0th index?

Reply:
Yes, it is! I fixed it :)